DTHPool.sol

```diff
@@ -0,0 +1,313 @@
+        }
+
+        balanceOf[msg.sender] += _amount;
+        totalPoolTokens += _amount;
+        Delegate(msg.sender, _amount);
+        return true;
+    }
+
+    function undelegateDAOTokens(uint _amount) returns (bool _success) {
+        if (_amount > balanceOf[msg.sender]) {
+            throw;
+        }
+
+        if (!dao.transfer(msg.sender, _amount)) {
+            throw;
+        }
+
+        balanceOf[msg.sender] -= _amount;
+        totalPoolTokens -= _amount;
+        Undelegate(msg.sender, _amount);
+        return true;
+    }
+
+    function setVoteIntention(
+        uint _proposalID,
+        bool _willVote,
+        bool _supportsProposal,
+        string _motivation
+    ) onlyDelegate returns (bool _success) {
+
+        ProposalStatus proposalStatus = proposalStatuses[_proposalID];
+
+        if (proposalStatus.voteSet) {
+            throw;
+        }
+
+        var (,,,votingDeadline, ,,,,newCurator) = dao.proposals(_proposalID);
+
+        if (votingDeadline < now || newCurator) {
+            throw;
+        }
+
+        proposalStatus.voteSet = true;
+        proposalStatus.willVote = _willVote;
+        proposalStatus.suportProposal = _supportsProposal;
+        proposalStatus.votingDeadline = votingDeadline;
+        proposalStatus.motivation = _motivation;
+
+        if (!_willVote) {
+            proposalStatus.executed = true;
+            VoteExecuted(_proposalID);
+        }
+
+        VoteIntentionSet(_proposalID, _willVote, _supportsProposal);
+
+        bool finalized = executeVote(_proposalID);
+
+        if (!finalized) {
+            pendingProposals[ pendingProposals.length++ ] = _proposalID;
+        }
+
+        return true;
+    }
+
+    function executeVote(uint _proposalID) returns (bool _finalized) {
+        ProposalStatus proposalStatus = proposalStatuses[_proposalID];
+
+        if (!proposalStatus.voteSet
+            || now > proposalStatus.votingDeadline
+            || !proposalStatus.willVote
+            || proposalStatus.executed) {
+            
+            return true;
+        }
+
+        if (now < proposalStatus.votingDeadline - maxTimeBlocked) {
+            return false;
+        }
+
+
+        dao.vote(_proposalID, proposalStatus.suportProposal);
+        proposalStatus.executed = true;
+        VoteExecuted(_proposalID);
+
+        return true;
+    }
+
+    function executeAllVotes() returns (bool _success) {
+        uint i;
+        for (i = 0; i < pendingProposals.length;) {
+            bool finalized = executeVote(pendingProposals[i]);
+            if (finalized) {
+                pendingProposals[i] = pendingProposals[pendingProposals.length - 1];
+                pendingProposals.length--;
+            } else {
+                i++;
+            }
+        }
+        return true;
+    }
+
+    function fixTokens() returns (bool _success) {
+        uint ownedTokens = dao.balanceOf(this);
+        if (ownedTokens < totalPoolTokens) {
+            throw;
```

Comment:
@colm: If you send tokens directly to DTHPool then ownedTokens > totalPoolTokens.
This method is precisely to recover those tokens and "send" them to the delegate. The delegate can return those tokens if he wants.